cppcache/src/TcrConnection.cpp

```diff
@@ -686,30 +682,27 @@ char* TcrConnection::readMessage(size_t* recvLen,
       reinterpret_cast<uint8_t*>(msg_header), HEADER_LENGTH);
   // ignore msgType
   input.readInt32();
-  msgLen = input.readInt32();
+  auto msgLen = input.readInt32();
   //  check that message length is valid.
-  if (!(msgLen > 0) && request == TcrMessage::GET_CLIENT_PR_METADATA) {
-    char* fullMessage;
-    *recvLen = HEADER_LENGTH + msgLen;
-    _GEODE_NEW(fullMessage, char[HEADER_LENGTH + msgLen]);
-    std::memcpy(fullMessage, msg_header, HEADER_LENGTH);
-    return fullMessage;
-    // exit(0);
+  if (msgLen <= 0 && request == TcrMessage::GET_CLIENT_PR_METADATA) {
+    auto fullMessage = make_unique<char[]>(sizeof(msg_header));
+    std::memcpy(fullMessage.get(), msg_header, sizeof(msg_header));
+    return {std::move(fullMessage), sizeof(msg_header), CONN_NOERR};
   }
 
-  // user has to delete this pointer
-  char* fullMessage;
-  *recvLen = HEADER_LENGTH + msgLen;
-  _GEODE_NEW(fullMessage, char[HEADER_LENGTH + msgLen]);
-  std::memcpy(fullMessage, msg_header, HEADER_LENGTH);
+  auto recvLen = HEADER_LENGTH + msgLen;
+  auto fullMessage = make_unique<char[]>(recvLen);
+  std::memcpy(fullMessage.get(), msg_header, HEADER_LENGTH);
 
   std::chrono::microseconds mesgBodyTimeout = receiveTimeoutSec;
   if (isNotificationMessage) {
     mesgBodyTimeout = receiveTimeoutSec * DEFAULT_TIMEOUT_RETRIES;
   }
-  error = receiveData(fullMessage + HEADER_LENGTH, msgLen, mesgBodyTimeout);
+  error =
+      receiveData(fullMessage.get() + HEADER_LENGTH, msgLen, mesgBodyTimeout);
   if (error != CONN_NOERR) {
-    delete[] fullMessage;
+    fullMessage.reset();
```

Comment:
Technically not necessary - dunno how attached you are to this line, I suppose it goes some small way towards documenting what we're doing here...

Reply:
Agreed, let the smart pointer do the work later. It doesn't do anything for us really to do it here explicitly.